File: src/main/java/com/box/Box.java
package com.box;

public abstract class Box {

    private float mLength;
    private float mWidth;
    private float mHeight;

    public Box(float length, float width, float height) {
        this.mLength = length;
        this.mWidth = width;
        this.mHeight = height;
    }

    abstract String BoxName();

    /**
     * 箱子是否能容納
     *
     * @param length
     * @param width
     * @param height
     * @return
     */
    public boolean validate(float length, float width, float height) {
        return !(length > mLength || width > mWidth || height > mHeight);
    }

}

File: src/main/kotlin/com/box/Box.kt
package com.box.kotlin

abstract class Box(private val mLength: Float, private val mWidth: Float, private val mHeight: Float) {
    fun validBox(length: Float, width: Float, height: Float): Boolean =
        !(length > mLength || width > mWidth || height > mHeight);

    abstract fun boxName(): String;
}

class Box3 : Box(23f, 14f, 13f) {
    override fun boxName(): String {
        return "Box3";
    }
}

class Box5 : Box(39.5f, 27.5f, 23f) {
    override fun boxName(): String {
        return "Box5";
    }
}